### PR TITLE
Remove mint when sendmint fail

### PR DESCRIPTION
**Pull request by thebevrishot**
closes https://github.com/zcoinofficial/zcoin/issues/660

### CODE REVIEW

File: src/exodus/wallet.cpp [hunk 1]
```diff
@@ -86,6 +86,11 @@ SigmaSpend Wallet::CreateSigmaSpend(PropertyId property, SigmaDenomination denom
     return SigmaSpend(SigmaMintId(*mint, DefaultSigmaParams), mint->chainState.group, anonimitySet.size(), proof);
 }
 
+bool Wallet::EraseSigmaMint(const SigmaMintId &id)
+{
+    return CWalletDB(walletFile).EraseExodusMint(id);
```
Comment (ultimaweapon) [inline src/exodus/wallet.cpp]: We should throw instead if it fail so it will more easy to use from the caller site.

File: src/exodus/rpctx.cpp [hunk 2]
```diff
@@ -1635,6 +1638,13 @@ UniValue exodus_sendmint(const UniValue& params, bool fHelp)
 
     // check error and return the txid (or raw hex depending on autocommit)
     if (result != 0) {
+        for (auto const &id : ids) {
+            try {
+                wallet->EraseSigmaMint(id);
+            } catch (std::runtime_error const &e) {
+                LogPrintf("%s : Fail to erase sigma mints, %s", __func__, e.what());
```
Comment (ultimaweapon) [inline src/exodus/rpctx.cpp]: Need to add line feed at the end of string.

File: qa/rpc-tests/exodus_sendmint.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+class ExodusSendMintTest(ExodusTestFramework):
+    def run_test(self):
+        super().run_test()
+
+        sigma_start_block = 500
+
+        self.nodes[0].generatetoaddress(100, self.addrs[0])
+        self.nodes[0].generate(sigma_start_block - self.nodes[0].getblockcount())
+
+        assert(sigma_start_block == self.nodes[0].getblockcount())
```
Comment (ultimaweapon) [inline qa/rpc-tests/exodus_sendmint.py]: Why do we use `assert` instead of `assert_equal`?

File: src/exodus/rpctx.cpp (hunk 2 above)
Comment (thebevrishot) [reply src/exodus/rpctx.cpp]: Nice catch

File: qa/rpc-tests/exodus_sendmint.py (hunk 3 above)
Comment (thebevrishot) [reply qa/rpc-tests/exodus_sendmint.py]: good idea to change to `assert_equal`.

**Pull request merged**